pkg/kds/context/context.go

```diff
@@ -246,6 +245,8 @@ func GlobalProvidedFilter(rm manager.ResourceManager, configs map[string]bool) r
 				zone_tokens.SigningKeyPrefix,
 			}...)
 		case r.Descriptor().KDSFlags.Has(core_model.GlobalToAllButOriginalZoneFlag):
+			// TODO (Icarus9913): replace the function by model.ZoneOfResource(r)
+			// Reference: https://github.com/kumahq/kuma/issues/10952
 			zoneTag := util.ZoneTag(r)
```

Comment:
@lobkovilya Can we put `ZoneTag` function refactor in another PR? For this PR, I just removed the KDS v1 source codes without any break or fix.

In another PR, I want to do such things:
1. Replace the `ZoneTag` function by `model.ZoneOfResource`
2. Disalow the ZoneIngress resource creation in Global
3. At present, if I have a multizone ENV with 2 zones and the zone-1 ZoneIngress resource would be synced to global and futhur synced to zone-2 due to the `ZoneTag` function. And this behavior would be reverted in the next PR but make it clear with the compatibility with those synced `zone-1 ZoneIngress resource` in zone-2

Reply:
Maybe we can get some refactor and move everything from v2 to normal package

Reply:
Sure, I don't think removing KDS v1 has something to do with zone validation of the resources on Global. So yes, we definitely should work on this in another PR.